src/main/java/com/minecolonies/tileentities/ScarecrowTileEntity.java

```diff
@@ -0,0 +1,123 @@
+package com.minecolonies.tileentities;
+
+import com.minecolonies.colony.Colony;
+import com.minecolonies.colony.ColonyManager;
+import com.minecolonies.inventory.InventoryField;
+import com.minecolonies.util.EntityUtils;
+import com.minecolonies.util.LanguageHandler;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.tileentity.TileEntityChest;
+import net.minecraft.world.World;
+
+import java.util.Random;
+
+/**
+ * The scarecrow tile entity to store extra data.
+ */
+public class ScarecrowTileEntity extends TileEntityChest
+{
+    /**
+     * NBTTag to store the type.
+     */
+    private static final String TAG_TYPE = "type";
+    /**
+     * Random generator.
+     */
+    private final Random random = new Random();
+    /**
+     * The inventory connected with the scarecrow.
+     */
+    private InventoryField inventoryField;
+    /**
+     * The type of the scarecrow.
+     */
+    private ScareCrowType  type;
+
+    /**
+     * Creates an instance of the tileEntity.
+     */
+    public ScarecrowTileEntity()
+    {
+        super();
+        World world = getWorld();
+        this.inventoryField = new InventoryField(LanguageHandler.getString("com.minecolonies.gui.inventory.scarecrow"), true);
+
+        if(world != null)
+        {
+            Colony colony = ColonyManager.getColony(world, pos);
+
+            if (colony != null && colony.getField(pos) == null)
+            {
+                Entity entity = EntityUtils.getEntityFromUUID(world, colony.getPermissions().getOwner());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'entity' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)